projects/ngrx.io/content/examples/store/src/app/state/allBooks.selectors.ts

```diff
@@ -0,0 +1,18 @@
+import { createSelector } from '@ngrx/store';
+import { AppState } from './state';
+import { Book } from '../book-list/books.service';
+
+export const selectBooks = createSelector(
+  (state: AppState) => state.books,
+  (books: Array<Book>) => books
+);
+
+export const selectCollectionIds = (state: AppState) => state.collection;
```

Comment:
Hi @sassypianist98, I come from #2706 and @timdeschryver told me that in this PR I may find some answers :)

I can see that both `selectBooks` and `selectCollectionIds` select a single piece of state, and yet one of them use createSelector, and the other one does not. Why is the reason to do so? My current understanding of createSelectors function is that you use is whenever you want to use memoization (which for me seems to be almost always), or when you want to have a selector for multiple pieces of state.

I can see 2 options for this file (they are mutually exclusive 😄):

* Option A

 `export const selectCollectionIds = (state: AppState) => state.collection;`

becomes

```
export const selectCollectionIds = createSelector(
  (state: AppState) => state.collection,
  (collectionIds: Array<string>) => collectionIds
);
```
Having that, `selectCollectionIds` benefits by receiving support for memoization. It can be exposed to and used by effects and components now.


* Option B

 `export const selectCollectionIds = (state: AppState) => state.collection;`

becomes

`const selectCollectionIds = (state: AppState) => state.collection;`

Having that, `selectCollectionIds` still does not utilize memoization, but it now considered 'private' selector - it is hidden from components and effects.

Reply:
For clarity, the top-level feature key should always use `createFeatureSelector`. 

```ts
export const selectCollectionState = createFeatureSelector<AppState, ReadonlyArray<string>('collection');
```

There is only a single piece of state here, but in normal practice, it would be an object that has multiple properties on it, each one building off the feature selector. We should keep the example straightforward while allowing for more guidance on usage going forward.